### PR TITLE
Mqtt rust keywords 4863 v4

### PR DESCRIPTION
**Pull request by catenacyber**
Link to ticket: https://redmine.openinfosecfoundation.org/issues/
https://redmine.openinfosecfoundation.org/issues/4683

Describe changes:
- mqtt: move keywords to pure rust

https://github.com/OISF/suricata/pull/11430 with review taken into account, and removing some unused includes in output-json-alert.c (especially the output-json-mqtt.h)

cc @satta

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 21467

File: src/detect-engine-helper.c [hunk 1]
```diff
@@ -81,6 +81,20 @@ int DetectHelperBufferMpmRegister(const char *name, const char *desc, AppProto a
     return DetectBufferTypeGetByName(name);
 }
 
+int DetectHelperMultiBufferMpmRegister(const char *name, const char *desc, AppProto alproto,
```
Comment (victorjulien) [inline src/detect-engine-helper.c:84]: these generic detect additions should be in their own commit

**Review by victorjulien — CHANGES_REQUESTED** [review]

File: rust/src/mqtt/detect.rs [hunk 2]
```diff
@@ -17,405 +17,1493 @@
...
+    if SigMatchAppendSMToList(de, s, G_MQTT_FLAGS_KW_ID, ctx, G_MQTT_FLAGS_BUFFER_ID).is_null() {
+        mqtt_flags_free(std::ptr::null_mut(), ctx);
+        return -1;
     }
     return 0;
 }
 
-#[no_mangle]
-pub extern "C" fn rs_mqtt_tx_unsuback_has_reason_code(tx: &MQTTTransaction, code: u8) -> u8 {
+extern "C" fn rs_mqtt_tx_has_flags(tx: &MQTTTransaction, ctx: &DetectUintData<u8>) -> c_int {
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply rust/src/mqtt/detect.rs:780]: nice find

File: rust/src/mqtt/detect.rs [hunk 3]
```diff
@@ -17,405 +17,1493 @@
...
+    let ctx = cast_pointer!(ctx, DetectUintData<u8>);
+    return rs_mqtt_tx_has_conn_flags(tx, ctx);
+}
+
+unsafe extern "C" fn mqtt_conn_flags_free(_de: *mut c_void, ctx: *mut c_void) {
+    let ctx = cast_pointer!(ctx, DetectUintData<u8>);
+    rs_detect_u8_free(ctx);
+}
+
+pub unsafe extern "C" fn mqtt_conn_willtopic_setup(
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply rust/src/mqtt/detect.rs:926]: Indeed, but these should not be public in the first place :-)

**Comment by catenacyber** [comment]
Continued in https://github.com/OISF/suricata/pull/11499

File: src/detect-engine-helper.c (hunk 1 above)
Comment (catenacyber) [reply src/detect-engine-helper.c:84]: ok
